app/src/org/commcare/utils/AppLifecycleUtils.java

```diff
@@ -50,6 +52,9 @@ public static void uninstall(ApplicationRecord record) {
         record.setStatus(ApplicationRecord.STATUS_DELETE_REQUESTED);
         ccInstance.getGlobalStorage(ApplicationRecord.class).write(record);
 
+        // cancel all Workmanager tasks for this app
+        WorkManager.getInstance(CommCareApplication.instance()).cancelAllWorkByTag(record.getApplicationId());
```

Comment:
This needs to occur before line 45 above, since once the instance is unseated we would start to experience issues if the worker continued until line 56.

I'd actually recommend that we add this "cancelAllWorkByTag" action directly to the `CCInstance.unseat()`method in general